### DISCUSSION TITLE
Golang and python auto-complete work differently

### DISCUSSION THREAD

**Discussion by llimllib** [hrsh7th/nvim-cmp#1056] [Q&A] [answered]
With [this config](https://gist.github.com/llimllib/85c5f3e7cbe59d455df0670b96a83a40), in a python buffer, nvim-cmp works like I would want. If I type `<symbol>.` and press tab, a list of complete options pops up but none are selected; the first time I press tab it goes to the first completion, and further tabs cycle through the list.

However, in a golang buffer, the completion opens to the middle of the list with the best completion selected; to actually select it I need to hit enter or `tab` followed by `shift-tab`.

How can I get the golang completion menu to work the same as the python completion menu?

Here's a screen recording illustrating the differences, in a clean vim setup:

https://user-images.githubusercontent.com/7150/176468849-e3275dbc-0de8-4abc-8fb3-f038f4810a06.mov

**Comment by ghost** [comment, marked as answer]
I had a similar problem with Rust. I think what you want is this: `:h cmp-config.preselect`. I added this to my cmp.setup and everything worked as I wanted it:
`
preselect = {
    cmp.PreselectMode.None
}
`

**Comment by llimllib** [comment]
That's exactly what I was looking for! Thanks
